salt/client/__init__.py

```diff
@@ -1935,7 +1931,83 @@ def pub(
             if not payload:
                 return payload
 
-        return {"jid": payload["load"]["jid"], "minions": payload["load"]["minions"]}
+        ret = {"jid": payload["load"]["jid"], "minions": payload["load"]["minions"]}
+        if "syndics" in payload:
+            ret["syndics"] = payload["syndics"]
+        return ret
+
+    @salt.ext.tornado.gen.coroutine
+    def arbitrary_pub_async(
+        self,
+        tgt,
+        payload,
+        tgt_type="glob",
+        timeout=5,
+        io_loop=None,
+        listen=True,
+        cmd=None,
+        **kwargs,
+    ):
+        """
+        Publish an arbitrary payload. Maybe this is what events do?
+        Publish an arbitrary payload. Maybe this is what events do?
```

Comment:
**suggestion** somehow I managed to duplicate a doc line here. Fix this docstring.